### PR TITLE
mqueue: simplify the mqueue reailize

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

mqueue: simplify the mqueue reailize

1. remove description management in mqueue, save code size
2. use i_ops instead of i_mqueue to remove the dup logic
3. add file_mq_xx for kernel use
4. Obey open nx_open file_open rules

## Impact

All places where use mq,
The developer should be know:

user space: mq_xx (mq_open returned is FD, not a structure)
kernel space: file_mq_xx

## Testing

Testing pass on sim

### CODE REVIEW

**Comment by GUIDINGLI** [comment]
This depends on https://github.com/apache/incubator-nuttx-apps/pull/544

**Comment by btashton** [comment]
The app PR is in I am restarting this build.

**Comment by xiaoxiang781216** [comment]
There is another issue when unlink is disabled, @liguiding please find a solution to fix it.

**Comment by GUIDINGLI** [comment]
Aslo depends on:
https://github.com/apache/incubator-nuttx-apps/pull/546

File: fs/mqueue/mq_unlink.c [hunk 1]
```diff
@@ -62,97 +62,43 @@
  *
  ****************************************************************************/
 
-int nxmq_unlink(FAR const char *mq_name)
+int file_mq_unlink(FAR const char *mq_name)
```
Comment (xiaoxiang781216) [inline fs/mqueue/mq_unlink.c:65]: don't need

File: fs/mqueue/mq_open.c [hunk 2]
```diff
@@ -235,6 +262,115 @@ int nxmq_open(FAR const char *mq_name, int oflags, mode_t mode,
...
+
+  va_start(ap, oflags);
+  ret = file_mq_vopen(&mq, mq_name, oflags, ap);
+  va_end(ap);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  *mqdes = files_allocate(mq.f_inode, mq.f_oflags, 0, 0);
```
Comment (xiaoxiang781216) [inline fs/mqueue/mq_open.c:364]: change the first 0 to mq.f_priv

File: fs/mqueue/mq_open.c [hunk 3]
```diff
@@ -266,31 +402,26 @@ int nxmq_open(FAR const char *mq_name, int oflags, mode_t mode,
...
-      mode = va_arg(ap, mode_t);
-      attr = va_arg(ap, FAR struct mq_attr *);
-      va_end(ap);
+      set_errno(-ret);
+      return ERROR;
     }
 
-  ret = nxmq_open(mq_name, oflags, mode, attr, &mqdes);
-  if (ret < 0)
+  mqdes = files_allocate(mq.f_inode, mq.f_oflags, 0, 0);
```
Comment (xiaoxiang781216) [inline fs/mqueue/mq_open.c:419]: change the first 0 to mq.f_priv

File: fs/mqueue/mq_unlink.c (hunk 1 above)
Comment (GUIDINGLI) [reply fs/mqueue/mq_unlink.c:65]: you mean f_pos ?
If so, done

File: fs/mqueue/mq_open.c (hunk 2 above)
Comment (GUIDINGLI) [reply fs/mqueue/mq_open.c]: f_pos ?
done

File: fs/mqueue/mq_open.c (hunk 3 above)
Comment (GUIDINGLI) [reply fs/mqueue/mq_open.c]: f_pos ?
done

File: fs/mqueue/mq_unlink.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/mqueue/mq_unlink.c:97]: Ok

**Review by xiaoxiang781216 — COMMENTED** [review]
LGTM.

**Review by jerpelea — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**